x/profiles/types/chain_link.go

```diff
@@ -0,0 +1,158 @@
+package types
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/ghodss/yaml"
+)
+
+// NewChainConfig is a constructor function for ChainConfig
+func NewChainConfig(name string, prefix string) ChainConfig {
+	return ChainConfig{
+		Name:             name,
+		Bech32AddrPrefix: prefix,
+	}
+}
+
+func (chainConfig ChainConfig) Validate() error {
+	if chainConfig.Name == "" {
+		return fmt.Errorf("chain name cannot be empty")
+	}
+	return nil
+}
+
+// ___________________________________________________________________________________________________________________
+
+// NewProof is a constructor function for Proof
+func NewProof(pubKey cryptotypes.PubKey, signature string, plainText string) Proof {
+	pubKeyAny, err := codectypes.NewAnyWithValue(pubKey)
+	if err != nil {
+		panic("failed to pack public key to any type")
+	}
+	return Proof{
+		PubKey:    pubKeyAny,
+		Signature: signature,
+		PlainText: plainText,
+	}
+}
+
+func (proof Proof) Validate() error {
+	_, err := hex.DecodeString(proof.Signature)
+	if err != nil {
+		return fmt.Errorf("failed to decode hex string of signature")
+	}
+
+	if strings.TrimSpace(proof.PlainText) == "" {
+		return fmt.Errorf("plain text can not be empty or blank")
+	}
+	return nil
```

Comment:
I think we should also make sure the provided pub key is a valid pub key (ie. it can be deserialized to a `cyrptotypes.PubKey` instance properly).

Reply:
If do so, it need to have a unpacker pass to the function because Proof.PubKey is Any type.
Then, when calling ValidateBasic() function to validate msg or packet, the Proof.Validate() can not be used directly cause ValidateBasic() does not allow to pass value.

Reply:
You're right. Can we at least check if it's a valid `hex` string (just like the signature)?

Reply:
@RiccardoM I think it can not be checked without unpacking, only can be checked if it is `nil` .

Reply:
You're right, I forgot it was an `Any` and not a hex string.